libs/zopengl/src/wrapper.zig

```diff
@@ -0,0 +1,2637 @@
+pub const RGB8 = bindings.RGB8;
+pub const RGB10 = bindings.RGB10;
+pub const RGB12 = bindings.RGB12;
+pub const RGB16 = bindings.RGB16;
+pub const RGBA2 = bindings.RGBA2;
+pub const RGBA4 = bindings.RGBA4;
+pub const RGB5_A1 = bindings.RGB5_A1;
+pub const RGBA8 = bindings.RGBA8;
+pub const RGB10_A2 = bindings.RGB10_A2;
+pub const RGBA12 = bindings.RGBA12;
+pub const RGBA16 = bindings.RGBA16;
+pub const VERTEX_ARRAY = bindings.VERTEX_ARRAY;
+
+// pub var drawArrays: *const fn (mode: Enum, first: Int, count: Sizei) callconv(.C) void = undefined;
+pub fn drawArrays(prim_type: PrimitiveType, first: usize, count: usize) void {
+    bindings.drawArrays(
+        @enumToInt(prim_type),
+        @intCast(Int, first),
+        @intCast(Sizei, count),
+    );
+}
+
+// pub var drawElements: *const fn (
+//     mode: Enum,
+//     count: Sizei,
+//     type: Enum,
+//     indices: ?*const anyopaque,
+// ) callconv(.C) void = undefined;
+// pub var polygonOffset: *const fn (factor: Float, units: Float) callconv(.C) void = undefined;
+// pub var copyTexImage1D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     internalformat: Enum,
+//     x: Int,
+//     y: Int,
+//     width: Sizei,
+//     border: Int,
+// ) callconv(.C) void = undefined;
+// pub var copyTexImage2D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     internalformat: Enum,
+//     x: Int,
+//     y: Int,
+//     width: Sizei,
+//     height: Sizei,
+//     border: Int,
+// ) callconv(.C) void = undefined;
+// pub var copyTexSubImage1D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     xoffset: Int,
+//     x: Int,
+//     y: Int,
+//     width: Sizei,
+// ) callconv(.C) void = undefined;
+// pub var copyTexSubImage2D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     xoffset: Int,
+//     yoffset: Int,
+//     x: Int,
+//     y: Int,
+//     width: Sizei,
+//     height: Sizei,
+// ) callconv(.C) void = undefined;
+// pub var texSubImage1D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     xoffset: Int,
+//     width: Sizei,
+//     format: Enum,
+//     type: Enum,
+//     pixels: ?*const anyopaque,
+// ) callconv(.C) void = undefined;
+// pub var texSubImage2D: *const fn (
+//     target: Enum,
+//     level: Int,
+//     xoffset: Int,
+//     yoffset: Int,
+//     width: Sizei,
+//     height: Sizei,
+//     format: Enum,
+//     type: Enum,
+//     pixels: ?*const anyopaque,
+// ) callconv(.C) void = undefined;
+
+// pub var bindTexture: *const fn (target: Enum, texture: Uint) callconv(.C) void = undefined;
+pub fn bindTexture(target: TextureTarget, texture: Texture) void {
+    bindings.bindTexture(@enumToInt(target), @bitCast(Uint, texture));
+}
+
+// pub var deleteTextures: *const fn (n: Sizei, textures: [*c]const Uint) callconv(.C) void = undefined;
+
+// pub var genTextures: *const fn (n: Sizei, textures: [*c]Uint) callconv(.C) void = undefined;
+pub fn genTextures(ptr_or_slice: anytype) void {
```

Comment:
This function is a bit complicated. Instead, I would prefer to have a function like below for each resource type (I think it is fine to generate only one resource name at a time):

```zig
pub fn genTexture() Texture {
   var name: Uint = undefined;
   bindings.genTextures(1, &name);
   return .{ ._ = name };
}
```

Reply:
I didn't like it much either. I'd like to be able to keep the multi object generation pattern. What about providing both `genTexture` and `genTextures` as in updated code?

Reply:
Make sense but `genTexture()` should return `Texture` so that a user can declare it `const`:
```zig
const texture = gl.genTexture();
```